src/peripheral/systick/mod.rs

```diff
@@ -22,40 +22,56 @@ mod reload_value;
 mod current_value;
 mod defs;
 
-use super::{Control, Register};
+use core::ops::{Deref, DerefMut};
 use volatile::Volatile;
 use self::defs::*;
+use self::control_status::CSR;
+use self::current_value::CVR;
+use self::reload_value::RVR;
+
+pub use self::control_status::ClockSource;
 
 /// Returns an instance of the SysTick to modify system tick behavior.
 pub fn systick() -> SysTick {
     SysTick::systick()
 }
 
 /// Control system tick behavior.
-#[derive(Copy, Clone)]
-pub struct SysTick {
-    mem_addr: *const u32,
-    csr: control_status::CSR,
-    rvr: reload_value::RVR,
-    cvr: current_value::CVR,
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
+#[doc(hidden)]
+pub struct RawSysTick {
+    csr: CSR,
+    rvr: RVR,
+    cvr: CVR,
```

Comment:
The SysTick timer has another register, the `CALIB` register right after the `CVR` register. It's not being used, but seems like we should put a placeholder there just to be aware of it.